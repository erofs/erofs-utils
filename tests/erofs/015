#!/bin/sh
# SPDX-License-Identifier: GPL-2.0+
#
# regression test for battach on full buffer block
#
seq=`basename $0`
seqres=$RESULT_DIR/$(echo $0 | awk '{print $((NF-1))"/"$NF}' FS="/")

# get standard environment, filters and checks
. "${srcdir}/common/rc"

cleanup()
{
	cd /
	rm -rf $tmp.*
}

_require_erofs
_require_fssum

# remove previous $seqres.full before test
rm -f $seqres.full

# real QA test starts here
echo "QA output created by $seq"

if [ -z $SCRATCH_DEV ]; then
	SCRATCH_DEV=$tmp/erofs_$seq.img
	rm -f SCRATCH_DEV
fi

localdir="$tmp/$seq"
rm -rf $localdir
mkdir -p $localdir

# collect files pending for verification
head -c 4032 /dev/urandom > $localdir/1
head -c 4095 /dev/urandom > $localdir/2
head -c 12345 /dev/urandom > $localdir/3  # arbitrary size

_scratch_mkfs -Eforce-inode-compact --ignore-mtime $localdir >> $seqres.full 2>&1 \
	|| _fail "failed to mkfs"
_scratch_mount 2>>$seqres.full

# verify the uncompressed image
FSSUM_OPTS="-MAC"
[ $FSTYP = "erofsfuse" ] && FSSUM_OPTS="${FSSUM_OPTS}T"

sum1=`$FSSUM_PROG $FSSUM_OPTS $localdir`
echo "$localdir checksum is $sum1" >>$seqres.full
sum2=`$FSSUM_PROG $FSSUM_OPTS $SCRATCH_MNT`
echo "$SCRATCH_MNT checksum is $sum2" >>$seqres.full

[ "x$sum1" = "x$sum2" ] || _fail "-->checkMD5 FAILED"
_scratch_unmount

echo Silence is golden
status=0
exit 0
