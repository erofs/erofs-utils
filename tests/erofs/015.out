QA output created by 015
Silence is golden
